simplyprint: Skip objects functionality

cjavad opened the pull request:
This PR consists of the following changes: 

- Restructered `exclude_object` handling to be sent over `job_info` as `skipped_objects`
- Append and track current job_id in `job_info` and `file_progress` messages
- Fixed old test-url (internal usage)

I decided to open up this PR against master instead of the current working dev-branch to avoid cluttering the repo with 2 PRs, but if we prefer doing it that way that is also fine.

Let me know if any changes are needed!

cjavad commented:
By choice this PRs scope is small, with a few lines of changes to power the new skip-objects functionality for klipper/moonraker.

CC/Review: @AlbertMN

AlbertMN reviewed (approved):
LGTM

Arksine commented:
Thanks.  I had a couple of comments regarding current object tracking.  If there is no need to update SP with the current object then we could probably just remove the `"curent_object":  ["current_object"]` field from the subscription request since it is invalid.

cjavad commented on moonraker/components/simplyprint.py:
```diff
@@ -539,7 +543,8 @@ async def _on_klippy_ready(self) -> None:
             "gcode_move": ["gcode_position"],
-            "exclude_object": None
+            "exclude_object": ["objects", "excluded_objects"],
+            "current_object": ["current_object"],
```
Totally correct, i was referencing some other code that access get_stats("current_object", "current_object"), but re-reading the moonraker docs it was just there haha.

We do *want* the current object if possible, so i refactored it a bit to include that.

cjavad commented on moonraker/components/simplyprint.py:
```diff
@@ -824,6 +828,8 @@ def send_status(self, status: Dict[str, Any], eventtime: float) -> None:
         if "gcode_move" in status:
             self.layer_detect.update(status["gcode_move"]["gcode_position"])
+        if "exclude_object" in status:
+            self._handle_exclude_object(status["exclude_object"])
```
I am not too familiar with the send_status / job timer difference in the moonraker component, i'll look into it, and will most likely use it if you deem it better! (As we do want to track the current object).

But based on my very weak assumptions about the _get_object_diff and the object cache, it is my understanding that my _handle_exclude_object function would only produce the changes?

cjavad commented:
Thank you @Arksine i've fixed up some of the things you mentioned and added some additional comments, i am quite interested in knowing more about this: https://github.com/Arksine/moonraker/pull/1014#discussion_r2312422195

cjavad commented:
Perfect, we'll do some final testing on our side and report back once we are done.

cjavad commented:
@Arksine these changes looks good to us as well, we are ready to merge.
